activemq-broker/src/main/java/org/apache/activemq/broker/jmx/OpenTypeSupport.java

```diff
@@ -156,19 +156,59 @@ protected void init() throws OpenDataException {
         }
 
         @Override
-        public Map<String, Object> getFields(Object o) throws OpenDataException {
+        public Map<String, Object> getFields(Object o) throws OpenDataException { //TODO(ken): consider throwing it here
```

Comment:
Following up on this `TODO`, I believe it makes sense to one of the following:
- throw the `JMSException` from this method. Methods which call `getFields` either throw `Exception` or bubble up to a method that calls `Exception`. Since `OpenDataException` is already thrown from `getFields` the calling classes already have exception handling logic for that.
- catch `JMSException` and throw `OpenDataException`. This simplifies the logic but the exception may not be completely accurate.

As discussed previously, setting the fields to `""` may incur unexpected behaviour.

Reply:
Another option is to throw a `JMSRuntimeException` from `ActiveMQMessage` instead of `JMSException`.
This exception doesn't need to be caught and handled.
_(I got the idea from JB's talk at reInvent)_

Reply:
That is a good idea. Let me double check the spec

Reply:
Actually, I can't throw JMSRuntimeException because the jakarta.jms.Message interface throw JMSException. But I will probably adopt the first suggestion.

Reply:
Actually I remove the entire message logic and decide against throwing exception for 3.7.6 and 3.7.9 because that's a rabbit hole. First of all, spec doesn't require us to do that (it says "may"). Second of all, this requirement is not possible to meet if application create the message object using its own implementation, i,e don't use `context#createMessage`to create the message. In that case, there's no way to "lock" the object.

It's the responsibility of the application to beware of that message should only be accessed after the send is completed and message objection doesn't support concurrency. The Jakarta messaging provider will not throw exception.